src/ethereum/spurious_dragon/state.py

```diff
@@ -384,18 +414,18 @@ def set_balance(account: Account) -> None:
 
 def touch_account(state: State, address: Address) -> None:
     """
-    Initializes an account to state.
+    Check if an account in the state is zero and if so, delete it.
 
     Parameters
     ----------
     state:
         The current state.
 
     address:
-        The address of the account that need to initialised.
+        The address of the account to check.
     """
-    if not account_exists(state, address):
-        set_account(state, address, EMPTY_ACCOUNT)
+    if get_account_optional(state, address) == EMPTY_ACCOUNT:
+        destroy_account(state, address)
```

Comment:
`touch_account` method now destroys an account? Maybe we should rename it.

Reply:
The operation `touch_account()` does is called "touching" everywhere else. We should give it that meaning for consistency.

I removed the old `touch_account()` as unneeded, but if you think it should be readded under a new name for clarity I'm happy to consider it.